### PR TITLE
BROKERS: Source-UpdateSourceAsync

### LINKED ISSUE #54: BROKERS: Source - UpdateSourceAsync

### PR DESCRIPTION
**Pull request by SlimAhmad**
Closes #54

### CODE REVIEW

File: GitFyle.Core.Api/Brokers/Storages/StorageBroker.Sources.cs [hunk 1]
```diff
@@ -14,5 +15,8 @@ internal partial class StorageBroker
 
         public ValueTask<Source> InsertSourceAsync(Source source) =>
             InsertAsync(source);
+
+        public async ValueTask<Source> UpdateSourceAsync(Source student) =>
+           await UpdateAsync(student);
```
Comment (cjdutoit) [inline GitFyle.Core.Api/Brokers/Storages/StorageBroker.Sources.cs:20]:
```suggestion
        public async ValueTask<Source> UpdateSourceAsync(Source source) =>
           await UpdateAsync(source);
```

File: GitFyle.Core.Api/Brokers/Storages/StorageBroker.cs [hunk 2]
```diff
@@ -42,6 +42,14 @@ private async ValueTask<T> InsertAsync<T>(T entity)
             return entity;
         }
 
+        private async ValueTask<T> UpdateAsync<T>(T @object)
+        {
+            this.Entry(@object).State = EntityState.Modified;
+            await this.SaveChangesAsync();
```
Comment (cjdutoit) [inline GitFyle.Core.Api/Brokers/Storages/StorageBroker.cs:48]: Remember to detach the object as we don't want the context to keep track of changes to an entity after it has been saved.

```suggestion
            await this.SaveChangesAsync();
            DetachSavedEntity(@object);
```

**Review by cjdutoit — CHANGES_REQUESTED** [review]
Resolve conflicts

File: GitFyle.Core.Api/Brokers/Loggings/ILoggingBroker.cs [hunk 3]
```diff
@@ -6,6 +6,7 @@
 
 namespace GitFyle.Core.Api.Brokers.Loggings
 {
+
```
Comment (cjdutoit) [inline GitFyle.Core.Api/Brokers/Loggings/ILoggingBroker.cs:9]: remove empty line

**Comment by kandarpgautam** [comment]
if this can be merged then I will have the UpdateAsync method.

**Pull request closed without merging**